packages/docusaurus-mdx-loader/src/index.js

```diff
@@ -60,9 +60,11 @@ module.exports = async function (fileString) {
     if (metadataPath) {
       // Add as dependency of this loader result so that we can
       // recompile if metadata is changed.
-      this.addDependency(metadataPath);
-      const metadata = await readFile(metadataPath, 'utf8');
-      exportStr += `\nexport const metadata = ${metadata};`;
+      if (existsSync(metadataPath)) {
```

Comment:
prefer async variant whenever it's not too annoying to apply like here

Reply:
I'm not totally sure of what this code do but I think we should not modify the loader there, as it's less fail-fast, and could silently ignore some important errors (like not having the createData and mdx loader paths in sync, not sure)

Instead we could modify the loader usage in pages like:

```
                    metadataPath: (mdxPath: string) => {
                      if (isIgnored(mdxPath)) { 
                        return undefined;
                      }
                      const aliasedSource = aliasedSitePath(mdxPath, siteDir);
                      return path.join(
                        dataDir,
                        `${docuHash(aliasedSource)}.json`,
                      );
                    },
```

Reply:
I did same earlier but then I tried importing file in docs and blogs plugins and failed in both.

> That looks good but a few things to change.
> 
> Wonder what's the behavior if I put frontmatter in a _file.md, maybe we should throw an exception explaining this is forbidden to the user. Also worth to add a test for that

Frontmatter is not in plugin control. It will require changes to mdx-loader

Reply:
will take a look on next review.

I think we could have a mdx loader option like `forbidFrontmatter: (mdxPath: string) => boolean`